kratos/includes/kratos_components.h

```diff
@@ -80,23 +81,19 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
 
     static void Add(std::string const& Name, TComponentType const& ThisComponent)
     {
+        // check if a different object was already registered with this name, since this is undefined behavior
+        auto it_comp =  msComponents.find(Name);
+        KRATOS_ERROR_IF(it_comp != msComponents.end() && typeid(*(it_comp->second)) != typeid(ThisComponent)) << "An object of different type was already registered with name \"" << Name << "\"!" << std::endl;
         msComponents.insert(ValueType(Name , &ThisComponent));
     }
 
     static TComponentType const& Get(std::string const& Name)
     {
         auto it_comp =  msComponents.find(Name);
-        if (it_comp != msComponents.end()) {
-            return *(it_comp->second);
-        }
 
-        std::stringstream err_msg;
-        err_msg << "The component \"" << Name << "\" is not registered!\n"
-                << "Maybe you need to import the application where it is defined?\n"
-                << "The following components of this type are registered:" << std::endl;
-        KratosComponents instance; // creating an instance for using "PrintData"
-        instance.PrintData(err_msg);
-        KRATOS_ERROR << err_msg.str() << std::endl;
+        KRATOS_DEBUG_ERROR_IF(it_comp == msComponents.end()) << GetMessageUnregisteredComponent(Name) << std::endl;
```

Comment:
@pooyan-dadvand now it works completely without `KRATOS_DEBUG`, also I think it is quite easy to read/understand, please have a lool